From 967cf687e9cf940eb642d60458869eabb0df6f4e Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Wed, 12 Feb 2020 22:33:49 -0700
Subject: [PATCH 24/31] Revert "objtool: Fail the kernel build on fatal errors"

Breaks clang. Revert while issue is being sorted out.

This reverts commit 644592d328370af4b3e027b7b1ae9f81613782d8.

Link: https://github.com/ClangBuiltLinux/linux/issues/872
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 tools/objtool/check.c | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/tools/objtool/check.c b/tools/objtool/check.c
index 2d52a40e6cb9..0825f86c5122 100644
--- a/tools/objtool/check.c
+++ b/tools/objtool/check.c
@@ -2533,14 +2533,8 @@ int check(const char *_objname, bool orc)
 out:
 	cleanup(&file);
 
-	if (ret < 0) {
-		/*
-		 *  Fatal error.  The binary is corrupt or otherwise broken in
-		 *  some way, or objtool itself is broken.  Fail the kernel
-		 *  build.
-		 */
-		return ret;
-	}
-
+	/* ignore warnings for now until we get all the code cleaned up */
+	if (ret || warnings)
+		return 0;
 	return 0;
 }
-- 
2.25.1

